sway-lsp/src/utils/keyword_docs.rs

```diff
@@ -0,0 +1,850 @@
+            /// ```sway
+            /// let thing1: u32 = 100;
+            /// let thing2 = 200 + thing1;
+            ///
+            /// let mut changing_thing = true;
+            /// changing_thing = false;
+            /// ```
+            ///
+            /// The pattern is most commonly a single variable, which means no pattern matching is done and
+            /// the expression given is bound to the variable. Apart from that, patterns used in `let` bindings
+            /// can be as complicated as needed, given that the pattern is exhaustive. The type of the pattern
+            /// is optionally given afterwards, but if left blank is automatically inferred by the compiler if possible.
+            ///
+            /// Variables in Sway are immutable by default, and require the `mut` keyword to be made mutable.
+            ///
+            /// Multiple variables can be defined with the same name, known as shadowing. This doesn't affect
+            /// the original variable in any way beyond being unable to directly access it beyond the point of
+            /// shadowing. It continues to remain in scope, getting dropped only when it falls out of scope.
+            /// Shadowed variables don't need to have the same type as the variables shadowing them.
+            ///
+            /// ```sway
+            /// let shadowing_example = true;
+            /// let shadowing_example: u32 = 123;
+            /// let shadowing_example = shadowing_example as u8;
+            /// ```
+            ///
+            /// Other places the `let` keyword is used include along with [`if`], in the form of `if let`
+            /// expressions. They're useful if the pattern being matched isn't exhaustive, such as with
+            /// enumerations.
+            mod let_keyword {}
+        };
+
+        let while_keyword: ItemMod = parse_quote! {
+            /// Loop while a condition is upheld.
+            ///
+            /// A `while` expression is used for predicate loops. The `while` expression runs the conditional
+            /// expression before running the loop body, then runs the loop body if the conditional
+            /// expression evaluates to `true`, or exits the loop otherwise.
+            ///
+            /// ```sway
+            /// let mut counter = 0;
+            ///
+            /// while counter < 10 {
+            ///     log(counter);
+            ///     counter += 1;
+            /// }
+            /// ```
+            ///
+            /// A `while` expression cannot break with a value and always evaluates to `()`.
+            ///
+            /// ```sway
+            /// let mut i = 1;
+            ///
+            /// while i < 100 {
+            ///     i *= 2;
+            ///     if i == 64 {
+            ///         break; // Exit when `i` is 64.
+            ///     }
+            /// }
+            /// ```
+            mod while_keyword {}
+        };
+
+        let true_keyword: ItemMod = parse_quote! {
```

Comment:
I don't see hover docs for true/false
<img width="293" alt="image" src="https://user-images.githubusercontent.com/47993817/216197524-a9aa4dd0-0586-4aa2-9be4-09d02306dfe0.png">

Reply:
Nice find, i'll open an issue for this as well.

Reply:
opened #3967